### ISSUE TITLE
C# compiler should report CS7003 (or similar) when call to IServiceProvider.GetService<T>() is missing generic type argument and is followed by another call

### ISSUE THREAD

**Issue by jpd30** [dotnet/roslyn#41779]
**Version Used**: 

Microsoft .NET Core SDK 3.1.100 (x64)
Microsoft Visual Studio Community 2019 16.4.2

**Steps to Reproduce**:

Build the attached project [FailedToEmitModule.zip](https://github.com/dotnet/roslyn/files/4225938/FailedToEmitModule.zip)
, or:

1. Create a new .NET Core console app
2. Add `<PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="3.1.2" />` to the `.csproj` file
3. Copy the following into `Program.cs`:

```cs
using System;
using Microsoft.Extensions.DependencyInjection;

namespace FailedToEmitModule
{
    public static class Program
    {
        public static void Main()
        {
            IServiceProvider serviceProvider = null;
            var foo = serviceProvider.GetService<>().Foo();
        }
    }
}
```

4. Build the project

**Expected Behavior**:

Build error along the lines of:

```
Program.cs(11,8,11,44): error CS7003: Unexpected use of an unbound generic name
```

This behaviour can be observed by removing the call to `Foo()`.

**Actual Behavior**:

Build error:

```
CSC : error CS7038: Failed to emit module 'FailedToEmitModule'
```

**Comment by jpd30** [comment]
CS0305 might be a better bet

#6236 is similar (and still *Open*, although the scenario described by that issue appears to have been resolved)

**Comment by RikkiGibson** [comment]
Here's a minimal repro:

```cs
interface IServiceProvider
{
    object GetService();
}

static class Program
{
    static T GetService<T>(this IServiceProvider obj) => default;
    
    static void M(IServiceProvider provider)
    {
        var service = provider.GetService<>().INVALID();
    }
}
```

I feel like I have seen a similar bug like this in the past year but I can't find it.

**Comment by RikkiGibson** [comment]
It feels like it's easy for analysis of calls like these to fall into a hole in conversions or overload resolution or wherever else. But these kinds of generic names are only allowed in 'typeof' expressions. Maybe when we bind such a name we could just check if it is contained in a typeof expression and if not give a diagnostic.

**Comment by kevinsun-dev** [comment]
Fixed in #45033, thanks for reporting the issue @jpd30!

**Comment by jpd30** [comment]
Thanks very much for the fix - really appreciate the effort you and the
team put in to ensure error messages are clear and lead straight to the
offending code, even in relatively obscure cases such as this.
